### PR TITLE
client scaling, buffered mouse clicks, trackpad camera rotation, misc fixes

### PR DESCRIPTION
**Pull request by conker-rsc**
This pull request contains the following enhancements:

- Client scaling via the nearest neighbor, bilinear interpolation, and bicubic interpolation algorithms
- Mouse click buffering to ensure synchronization between custom overlay clicks and the game loop
- Enabling horizontal trackpad swipes to rotate the camera (requires a compatible trackpad)
- Other minor fixes and general cleanup

### CODE REVIEW

File: src/Client/Settings.java [hunk 1]
```diff
@@ -2414,13 +2473,34 @@ public static void loadKeybinds(Properties props) {
...
+
+    resolveNewDefaults();
+  }
+
+  /**
+   * Resolves conflicting keybindings when a new merge has decided to steal and replace a
+   * pre-existing keybinding.
+   */
+  private static void resolveNewDefaults() {
+    // show_player_controls was rebound from alt-up to alt-c
```
[the remark this answers is not in the document]
Comment (conker-rsc) [reply src/Client/Settings.java]: Added a date to the comment

File: src/Client/ScaledWindow.java [hunk 2]
```diff
@@ -0,0 +1,900 @@
...
+    setFocusTraversalKeysEnabled(false);
+
+    // Add window listeners
+    addWindowListener(this);
+    addComponentListener(this);
+    addFocusListener(this);
+    addKeyListener(this);
+
+    // Allows drag-n-dropping a replay onto the game window
+    setDropTarget(ReplayQueue.dropReplays);
```
[the remark this answers is not in the document]
Comment (conker-rsc) [reply src/Client/ScaledWindow.java:104]: Yeah I think we had talked about this previously. We should probably log a github issue about it

File: src/Client/JClassPatcher.java [hunk 3]
```diff
@@ -1057,6 +1057,41 @@ else if (invoke.name.equals("addKeyListener"))
           }
         }
       }
+
+      // draw loading screen
```
[the remark this answers is not in the document]
Comment (conker-rsc) [reply src/Client/JClassPatcher.java:1061]: Meant to revisit this and completely forgot about it, thanks. Should be fixed now

**Comment by Hubcapp** [comment]
LGTM Thanks immensely for your contribution, hard work, patience, perseverance, and teamwork, in getting this built and tested.

**Pull request merged**